Runtime/Core/ApiGatewayManagement/AmazonGameLiftClientWrapper.cs

```diff
@@ -42,17 +56,143 @@ public async Task<SearchGameSessionsResponse> SearchGameSessions(SearchGameSessi
+            return _amazonGameLiftClient.CreateFleetAsync(request);
+        }
+        
+        public async Task<string> GenerateAuthToken(string computeName, string fleetId)
+        {
+            try
+            {
+                var computeAuthTokenRequest = new GetComputeAuthTokenRequest
+                {
+                    ComputeName = computeName,
+                    FleetId = fleetId
+                };
+                var computeAuthTokenResponse = await GetComputeAuthToken(computeAuthTokenRequest);
+                return computeAuthTokenResponse.AuthToken;
+            }
+            catch (Exception e)
             {
-                ServiceURL = localEndpoint
-            });
+                Console.WriteLine(e);
+                throw;
+            }
         }
 
-        public async Task<DescribeGameSessionsResponse> DescribeGameSessions(DescribeGameSessionsRequest request)
+        public async Task<string> RegisterCompute(string computeName, string fleetId, string fleetLocation, string ipAddress)
         {
-            return await _amazonGameLiftClient.DescribeGameSessionsAsync(request);
+            try
+            {
+                var registerComputeRequest = new RegisterComputeRequest()
+                {
+                    ComputeName = computeName,
+                    FleetId = fleetId,
+                    IpAddress = ipAddress,
+                    Location = fleetLocation
+                };
+                var registerComputeResponse = await RegisterCompute(registerComputeRequest);
+                
+                return registerComputeResponse.Compute.GameLiftServiceSdkEndpoint;
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+                throw;
+            }
+        }
+
+        public async Task CreateCustomLocationIfNotExists(string fleetLocation)
+        {
+            try
+            {
+                var listLocationsResponse = await ListLocations(new ListLocationsRequest
+                {
+                    Filters = new List<string>{ "CUSTOM" }
+                });
+                
+                var foundLocation = listLocationsResponse.Locations.FirstOrDefault(l => l.LocationName.ToString() == fleetLocation);
+
+                if (foundLocation == null)
+                {
+                    var createLocationResponse = await CreateLocation(new CreateLocationRequest()
+                    {
+                        LocationName = fleetLocation
+                    });
+                    
+                    if (createLocationResponse.HttpStatusCode == HttpStatusCode.OK)
+                    {
+                        Console.WriteLine($"Created Custom Location {fleetLocation}");
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
```

Comment:
Should we silently fail if there's an exception?

Reply:
Will be changed to produce UI errors.